src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -218,15 +225,9 @@ def __apply_private_op(
         Raises:
             ValueError: If "op_str" not supported.
         """
-        y = ReplicatedSharedTensor.sanity_checks(self, y, op_str)
+        y, session_vals = ReplicatedSharedTensor.sanity_checks(self, y)
```

Comment:
MAYBE : Here , we could have , if we have  op_str in `{mul,matmul` we could pass `sanity_checks(...,falcon_check=True)`

Reply:
`falcon_check` or `parties_check` or some name which suits best for the functionality.